app/main.py

```diff
@@ -1,2 +1,66 @@
+from typing import Any, Optional
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.length: int = 0
+        self.capacity: int = 8
+        self.hash_table: list[Optional[tuple]] = [None] * self.capacity
+
+    def _get_key_value(self, key: Any) -> int | None:
+        index = hash(key) % self.capacity
+        original_index = index
+
+        while self.hash_table[index] is not None:
+            if self.hash_table[index][0] == key:
+                return self.hash_table[index][2]
+            index = (index + 1) % self.capacity
+
+            if index == original_index:
+                break
+
+        return None
+
+    def _set_key_value(self, key: Any, value: Any) -> int | None:
```

Comment:
The use of '|' for type hints is only supported in Python 3.10 and later. If compatibility with earlier versions is needed, consider using 'Optional[int]' instead of 'int | None'.